app/cdap/components/WranglerGrid/SelectColumnPanel/utils.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  IHeaderNamesList,
+  IMultipleSelectedFunctionDetail,
+} from 'components/WranglerGrid/SelectColumnPanel/types';
+import { MULTI_SELECTION_COLUMN } from 'components/WranglerGrid/SelectColumnPanel/constants';
+
+export const getColumnsSupportedType = (
+  transformationDataType: string[],
+  columnsList: IHeaderNamesList[]
+) => {
+  return transformationDataType?.length > 0 && transformationDataType?.includes('all')
```

Comment:
transformationDataType? -- why is the optional chaining necessary when its required in the type definition of the function?

Reply:
same thing with columnslist

Reply:
Removed optional chaining wherever type is declared